In [40]:
import pandas as pd

In [41]:
df_sleep = pd.read_csv("Time Americans Spend Sleeping.csv")

In [42]:
df_sleep.shape

(945, 9)

In [43]:
df_sleep.columns

Index(['index', 'Year', 'Period', 'Avg hrs per day sleeping', 'Standard Error',
       'Type of Days', 'Age Group', 'Activity', 'Sex'],
      dtype='object')

In [44]:
display(df_sleep.head())

index  Year  Period  Avg hrs per day sleeping  Standard Error Type of Days  \
0      0  2003  Annual                      8.57           0.018     All days   
1      1  2004  Annual                      8.55           0.026     All days   
2      2  2005  Annual                      8.62           0.023     All days   
3      3  2006  Annual                      8.63           0.024     All days   
4      4  2007  Annual                      8.57           0.024     All days   

           Age Group  Activity   Sex  
0  15 years and over  Sleeping  Both  
1  15 years and over  Sleeping  Both  
2  15 years and over  Sleeping  Both  
3  15 years and over  Sleeping  Both  
4  15 years and over  Sleeping  Both

### Dataset Description: Time Use in Sleep by Americans

This dataset shows how much time Americans spend sleeping per day, segmented by **year, gender, age group, and type of day** (e.g., weekend vs weekday).

Each row represents a demographic segment for a specific year and day type.

**Columns:**
- Year, Period, Type of Days
- Avg hrs per day sleeping, Standard Error
- Age Group, Sex, Activity

**Shape:** 945 rows × 9 columns

This dataset is useful for studying behavioral sleep trends in the U.S. population and comparing sleep patterns across different groups over time.


### 😴 Data Type Classification: Time Americans Spend Sleeping

**🔢 Quantitative columns:**
- Avg hrs per day sleeping  
- Standard Error  

These values reflect the average daily sleep time and its statistical variation.

**👥 Qualitative columns:**
- Year  
- Period  
- Type of Days  
- Age Group  
- Activity  
- Sex  

These fields describe the structure of the observation: time frame, demographic group, and the kind of day (weekday vs weekend).


In [45]:
df_sleep["Activity"].unique()

array(['Sleeping'], dtype=object)

In [46]:
df_sleep["Age Group"].unique()

array(['15 years and over', '15 to 24 years', '25 to 34 years',
       '35 to 44 years', '45 to 54 years', '55 to 64 years',
       '65 years and over'], dtype=object)

In [47]:
df_sleep["Type of Days"].unique()

array(['All days', 'Nonholiday weekdays', 'Weekend days and holidays'],
      dtype=object)

In [48]:
df_sleep["Sex"].unique()

array(['Both', 'Men', 'Women'], dtype=object)

### 🔍 Observational Insight: Time Americans Spend Sleeping

After exploring the dataset more closely, I discovered that it offers more demographic granularity than initially expected.

**Relevant columns with diverse values:**
- `Sex`: Both, Men, Women  
- `Age Group`: 7 categories, including detailed age ranges from "15 to 24 years" to "65 years and over"  
- `Type of Days`: All days, Nonholiday weekdays, Weekend days and holidays  

**Interesting opportunities for analysis:**
- Compare **average sleep duration by gender** to understand if men or women tend to sleep more.
- Analyze **age-based sleep patterns** — e.g., do older people sleep more than younger ones?
- Evaluate **differences between weekdays and weekends**, to detect possible sleep compensation behavior.

**Why this matters:**
- Can help assess whether people are getting enough rest on workdays.
- May reveal patterns where individuals either oversleep or undersleep during weekends.
- Could highlight lifestyle trade-offs — such as sacrificing sleep to pursue productivity or leisure.

**Potential applications:**
- **Healthcare research**: to better understand how sleep varies across age and lifestyle.
- **Public policy and awareness**: campaigns promoting healthier sleep routines.
- **Product development**: for sleep-related solutions (e.g., mattresses, apps, supplements) targeting sle
